core/shared/src/main/scala/zio/clock/package.scala

```diff
@@ -27,18 +27,41 @@ package object clock {
 
   object Clock extends PlatformSpecific with Serializable {
     trait Service extends Serializable {
+
       def currentTime(unit: TimeUnit): UIO[Long]
+
+      // Could be UIO. We keep IO to preserve binary compatibility.
```

Comment:
Can this really be `UIO`? I know we changed this at what point because we observed that it really did throw `DateTimeException` when the system time zone wasn't in the list that was supported by the `java-time` implementation.

Reply:
The previous implementation used `ZoneId.systemDefault` in `currentDateTime`, which may throw a `DateTimeException`. The new implementation uses `Instant.now`, which does not throw according to the docs.